### PR TITLE
MAILBOX-342 (slightly) Speedup Cassandra mapper tests

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/utils/GuiceUtils.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.util.Modules;
+
+public class GuiceUtils {
+    public static Injector testInjector(CassandraCluster cluster) {
+        return Guice.createInjector(
+            Modules.combine(
+                binder -> binder.bind(MessageId.Factory.class).toInstance(new CassandraMessageId.Factory()),
+                binder -> binder.bind(org.apache.james.blob.api.BlobId.Factory.class).toInstance(new CassandraBlobId.Factory()),
```
Comment (rouazana) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/utils/GuiceUtils.java]: unneeded fqdn?

File: mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java [hunk 2]
```diff
@@ -1,164 +1,139 @@
...
+import javax.mail.Flags;
+
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.MailboxPath;
+
+/**
+ * A mock mailbox manager.
```
Comment (rouazana) [inline mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java]: no more

File: mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java [hunk 3]
```diff
@@ -1,164 +1,139 @@
...
+                        String subSubfolderName = subFolderName + ".SUBSUB_FOLDER_" + l;
+                        mailboxPath = MailboxPath.forUser(user, subSubfolderName);
+                        createMailbox(mailboxManager, mailboxSession, mailboxPath);
+
+                    }
+                        
+                }
+
+                mailboxManager.logout(mailboxSession, true);
+
```
Comment (rouazana) [inline mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java]: too much spaces?

**Comment by aduprat** [comment]
```
Tests in error: 
  ToRepositoryTest.userShouldBeAbleToAccessReprocessedMails:133 » ConditionTimeout
```
test this please

File: mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+     * Number of Messages per Mailbox to be created in the Mailbox Manager.
+     */
+    public static final int MESSAGE_PER_MAILBOX_COUNT = 3;
+    
+    /**
+     * Utility method to feed the Mailbox Manager with a number of 
+     * mailboxes and messages per mailbox.
+     */
+    public static void feedMailboxManager(MailboxManager mailboxManager) throws MailboxException, UnsupportedEncodingException {
+        for (int i = 0; i < DOMAIN_COUNT; i++) {
```
Comment (aduprat) [inline mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java]: No way to use `IntStream.range(0, DOMAIN_COUNT)`?

Comment (rouazana) [reply mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java]: I guess it's to avoid sneakyThrow()

Comment (chibenwa) [reply mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java]: No it's just that this code is uggly, I did not wanted to spend too much time reworking it...

Comment (trantienduchn) [reply mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java]: Here is the alternative, but look not better :))
```java
        IntStream.range(0, DOMAIN_COUNT).boxed()
            .flatMap(domainCount -> IntStream
                .range(0, USER_COUNT)
                .mapToObj(userCount -> Pair.of(domainCount, userCount)))
            .forEach(Throwing
                .<Pair<Integer, Integer>>consumer(pair -> provisionUser(mailboxManager, "user" + pair.getKey() + "@localhost" + pair.getValue()))
                .sneakyThrow());
```

**Comment by chibenwa** [comment]
Before:

```
[INFO] Apache James :: Mailbox :: Cassandra ............... SUCCESS [27:20 min]
```

After:

```
[INFO] Apache James :: Mailbox :: Cassandra ............... SUCCESS [21:08 min]
```

Gain: 6 min 12s ... Modest 3.64% mvn time enhancement.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**